cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -114,6 +114,14 @@ func (i *PipelineManager) preConfigureCheck(ctx context.Context, infraOptions pr
 		)
 	}
 
+	// Required authorized role assignments
+	checkRoles := []string{"Owner", "User Access Administrator"}
+	// Check permission on role assignment
+	err = i.azCli.CheckRoleAssignments(ctx, i.Environment.GetSubscriptionId(), checkRoles)
+	if err != nil {
+		return configurationWasUpdated, err
+	}
```

Comment:
In my opinion we should NOT be enforcing that the service principal has `Owner` or `User Access Administrator` for `azd pipeline config` to be run.  

The required roles solely depends on the underlying apps requirements.  Yes - anything that performs role assignments in the IaC will require these but many apps do not.

I would instead invest the time in providing a better error message when we see these types of ARM deployment failures and instruct the user what to do.  This is easily correct by the user by providing the `--principal-role owner` argument to the `azd pipeline config` command.

Consider adding similar troubleshooting tips in the template readme if it doesn't exist already.
https://github.com/azure-samples/todo-nodejs-mongo-aks#troubleshootingknown-issues